modules/dbnd-airflow/src/dbnd_airflow/scheduler/single_dag_run_job.py

```diff
@@ -766,3 +804,58 @@ def heartbeat_callback(self, session=None):
             # self.executor.terminate()
             return
         self.terminating = True
+
+
+@provide_session
+def find_and_kill_zombies(args, session=None):
+    """
+    Find zombie DagRuns and TaskInstances that are in not end state and for which related
+    BackfillJob has failed.
+    """
+    seconds_from_last_heartbeat = 30
+    last_expected_heartbeat = timezone.utcnow() - datetime.timedelta(
+        seconds=seconds_from_last_heartbeat
+    )
+    logger.info(
+        "Cleaning zombie tasks with heartbeat older than: %s", last_expected_heartbeat,
+    )
+
+    # Select BackfillJob that failed or are still running
+    # but have stalled heartbeat
+    failed_jobs_id_qry = (
+        session.query(BaseJob.id)
+        .filter(BaseJob.job_type == "BackfillJob")
+        .filter(
+            or_(
+                BaseJob.state == State.FAILED,
```

Comment:
we will start to scan all Failed jobs all the time., I think that will make it feature very heavy
let's for now take care of Running Dagruns only

Reply:
oh, now I understand, that's only for failed Job (ids